crates/voicevox_core_c_api/tests/e2e/testcases/user_dict_load.rs

```diff
@@ -0,0 +1,169 @@
+// ユーザー辞書の登録によって読みが変化することを確認するテスト。
+// 辞書ロード前後でAudioQueryのkanaが変化するかどうかで確認する。
+
+use crate::symbols::VoicevoxInitializeOptions;
+use assert_cmd::assert::AssertResult;
+use once_cell::sync::Lazy;
+use std::ffi::{CStr, CString};
+use std::mem::MaybeUninit;
+use test_util::OPEN_JTALK_DIC_DIR;
+use voicevox_core::result_code::VoicevoxResultCode;
+
+use libloading::Library;
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    assert_cdylib::{self, case, Utf8Output},
+    snapshots,
+    symbols::{Symbols, VoicevoxAccelerationMode, VoicevoxUserDictWordType},
+};
+
+macro_rules! cstr {
+    ($s:literal $(,)?) => {
+        CStr::from_bytes_with_nul(concat!($s, '\0').as_ref()).unwrap()
+    };
+}
+
+case!(TestCase);
+
+#[derive(Serialize, Deserialize)]
+struct TestCase;
+
+#[typetag::serde(name = "user_dict_load")]
+impl assert_cdylib::TestCase for TestCase {
+    unsafe fn exec(&self, lib: &Library) -> anyhow::Result<()> {
+        let Symbols {
+            voicevox_user_dict_word_make,
+            voicevox_user_dict_new,
+            voicevox_user_dict_add_word,
+            voicevox_user_dict_delete,
+            voicevox_default_initialize_options,
+            voicevox_default_audio_query_options,
+            voicevox_open_jtalk_rc_new,
+            voicevox_open_jtalk_rc_load_user_dict,
+            voicevox_open_jtalk_rc_delete,
+            voicevox_voice_model_new_from_path,
+            voicevox_voice_model_delete,
+            voicevox_synthesizer_new_with_initialize,
+            voicevox_synthesizer_delete,
+            voicevox_synthesizer_load_voice_model,
+            voicevox_synthesizer_audio_query,
+            ..
+        } = Symbols::new(lib)?;
+
+        let dict = voicevox_user_dict_new();
+
+        let mut word_uuid = [0u8; 16];
+
+        let word = {
+            let mut word = voicevox_user_dict_word_make(
+                CString::new("this_word_should_not_exist_in_default_dictionary")
+                    .unwrap()
+                    .into_raw(),
+                CString::new("アイウエオ").unwrap().into_raw(),
```

Comment:
```suggestion
                cstr!("this_word_should_not_exist_in_default_dictionary").as_ptr(),
                cstr!("アイウエオ").as_ptr(),
```

(`cstr!`も利用頻度が大きくなるようなら手書きじゃなくて[cstrクレート](https://crates.io/crates/cstr)を使う感じにしましょうかね...)